wsr-tic-tac-toe/src/__tests__/game-board.test.tsx

```diff
@@ -0,0 +1,226 @@
+            dataHook: CREATE_GAME_BUTTON,
+        });
+
+        await createGameButtonDriver.click();
+        const inputDriver = InputWithLabelTestkit({ wrapper: wrapper, dataHook: BOARD_SIZE_INPUT });
+
+        await inputDriver.enterText('2');
+        const messageBoxDriver = MessageBoxFunctionalLayoutTestkit({
+            wrapper: wrapper,
+            dataHook: CREATE_GAME_MESSAGE_BOX,
+        });
+        await messageBoxDriver.clickOnConfirmationButton();
+
+        const resetGameButtonDriver = ButtonTestkit({
+            wrapper: wrapper,
+            dataHook: RESET_GAME_BUTTON,
+        });
+        expect(await resetGameButtonDriver.exists()).toBeTruthy();
+    });
+
+    it('should show 4 squares after creating a game with boardSize: 2', async () => {
+        const createGameButtonDriver = ButtonTestkit({
+            wrapper: wrapper,
+            dataHook: CREATE_GAME_BUTTON,
+        });
+
+        await createGameButtonDriver.click();
+        const inputDriver = InputWithLabelTestkit({ wrapper: wrapper, dataHook: BOARD_SIZE_INPUT });
+
+        await inputDriver.enterText('2');
+        const messageBoxDriver = MessageBoxFunctionalLayoutTestkit({
+            wrapper: wrapper,
+            dataHook: CREATE_GAME_MESSAGE_BOX,
+        });
+        await messageBoxDriver.clickOnConfirmationButton();
+
+        const square0 = document.getElementById('0');
+        const square1 = document.getElementById('1');
+        const square2 = document.getElementById('2');
+        const square3 = document.getElementById('3');
+        expect(!!square0 && !!square1 && !!square2 && !!square3).toBeTruthy();
+    });
+
+    it('should show ONLY 4 squares after creating a game with boardSize: 2', async () => {
+        const createGameButtonDriver = ButtonTestkit({
+            wrapper: wrapper,
+            dataHook: CREATE_GAME_BUTTON,
+        });
+
+        await createGameButtonDriver.click();
+        const inputDriver = InputWithLabelTestkit({ wrapper: wrapper, dataHook: BOARD_SIZE_INPUT });
+
+        await inputDriver.enterText('2');
+        const messageBoxDriver = MessageBoxFunctionalLayoutTestkit({
+            wrapper: wrapper,
+            dataHook: CREATE_GAME_MESSAGE_BOX,
+        });
+        await messageBoxDriver.clickOnConfirmationButton();
+        const squareCount = document.evaluate(
+            'count(//div[contains(@class,"square")])',
+            document,
+            null,
+            XPathResult.ANY_TYPE,
+            null
+        ).numberValue;
+        expect(squareCount).toEqual(4);
+    });
+
+    it('should change turn order after clicking a square', async () => {
+        const createGameButtonDriver = ButtonTestkit({
+            wrapper: wrapper,
+            dataHook: CREATE_GAME_BUTTON,
+        });
+
+        await createGameButtonDriver.click();
+        const inputDriver = InputWithLabelTestkit({ wrapper: wrapper, dataHook: BOARD_SIZE_INPUT });
+
+        await inputDriver.enterText('2');
```

Comment:
there's a lot of repeating setup in later tests (clicking button, entering '2', closing modal)

You can replace this with a store that has a different initial state, instead of the default one